db/db_test2.cc

```diff
@@ -5439,6 +5439,108 @@ TEST_F(DBTest2, AutoPrefixMode1) {
     ASSERT_EQ("a1", iterator->key().ToString());
   }
 }
+
+class TestEnvWrapper : public EnvWrapper {
+ public:
+  TestEnvWrapper(Env* _target) : EnvWrapper(_target) {}
+  Status NewWritableFile(const std::string& fname,
+                         std::unique_ptr<WritableFile>* result,
+                         const EnvOptions& env_opts) override {
+    Status s = target()->FileExists(fname);
+    EXPECT_TRUE(s.IsNotFound()) << fname << " already exists.";
+    return target()->NewWritableFile(fname, result, env_opts);
+  }
+};
+
+class RenameCurrentTest : public DBTestBase,
+                          public testing::WithParamInterface<std::string> {
+ public:
+  RenameCurrentTest()
+      : DBTestBase("rename_current_test", /*env_do_fsync=*/true),
+        sync_point_(GetParam()) {}
+
```

Comment:
Thanks. I think it may be more complex than this.
`~DBTestBase()` will be called later which will `delete` the `db` pointer. At that time, the `TestEnv` object created in `RenameCurrentTest` is already destroyed, even if it is a `std::unique_ptr` in `RenameCurrentTest`. Therefore, we have to close the db anyway.

Reply:
You could address issue by doing a Close() in the Rename test destructor.  It might be a minor issue, but I can see someone adding a test in the future that fails in a strange way because they forgot to call Close() at the end.

Reply:
Make sense, will update.

Reply:
Done

Reply:
We ended up moving this logic to `SpecialEnv`.